libraries/Microsoft.Bot.Builder.Ai/TranslationMiddleware.cs

```diff
@@ -40,48 +46,33 @@ public TranslationMiddleware(HttpClient httpClient, string[] nativeLanguages, st
                 if (!String.IsNullOrWhiteSpace(message.Text))
                 {
                     // determine the language we are using for this conversation
-                    var sourceLanguage = "en"; // context.Conversation.Data["Language"]?.ToString() ?? this.nativeLanguages.FirstOrDefault() ?? "en";
-
-                    var translationContext = new TranslationContext
+                    var sourceLanguage = await Task.Run(() => translator.Detect(message.Text)); // context.Conversation.Data["Language"]?.ToString() ?? this.nativeLanguages.FirstOrDefault() ?? "en";
```

Comment:
>wait Task.Run(() => translator.Detect(message.Text)); [](start = 42, length = 53)

we should let the developer provide a callback so that they can specifically implement their own language detection logic. This will let them look at something stored in conversation or user state for instance.

Reply:
Updated now the developer can provide his callback or he can use Microsoft API language detection